TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -316,7 +316,7 @@ public virtual DataType PrimitiveDataType
             {
                 if (_primitiveDataType != null) return _primitiveDataType;
                 if (this.Picture != null) //Get DataType based on Picture clause
-                    _primitiveDataType = DataType.Create(this.Picture.Value);
+                    _primitiveDataType = DataType.Create(this._CommonDataDesc.PictureValidationResult);
```

Comment:
So this is `this._CommonDataDesc.DataType` right ? Do we have to create a new instance here ?

Reply:
No, because in the case of a Typedef, the DataType reflects the Type and not the picture clause. So we have to create a new instance.
I rewrote the code a little bit to explain that.